### PR TITLE
Split Celery integration tests to "celery" and "redis"

### LINKED ISSUE #34032: Split Celery integration tests in to "redis", "rabbitmq", "sqlalchemy"
### Body

Our celery integration tests assume only "redis" as both broker and result backend, however, we should be able to relatively easily add a few more integration configurations there. 

That would - for example - make it possible to detect earlied that Celery 5.3.3 and 5.3.2 is broken when SQLalchemy result backend is used https://github.com/celery/celery/issues/8470 - resulting in https://github.com/apache/airflow/pull/34031  and then it would not even make it into our constraints and our "canary" builds could detect it.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by shahar1**
closes: #34032

This PR takes care of the following:
- Splitting integration tests specific to the `redis` provider into their own category, as it makes testing more precise and avoids unnecessary initialization of the `rabbitmq` service.
- Running all celery tests with all possible brokers (currently only `redis` and `rabbitmq`). If a broker exists as an independent provider that has its own integration tests, we could import the service from the `integration-<PROVIDER>.yml` into the `integration-celery.yml` using docker-compose [extension](https://docs.docker.com/compose/multiple-compose-files/extends/) mechanism.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/breeze/src/airflow_breeze/params/shell_params.py [hunk 1]
```diff
@@ -539,8 +527,6 @@ def env_variables_for_docker_commands(self) -> dict[str, str]:
         _set_var(_env, "PYTHON_MAJOR_MINOR_VERSION", self.python)
         _set_var(_env, "QUIET", self.quiet)
         _set_var(_env, "REDIS_HOST_PORT", None, REDIS_HOST_PORT)
-        _set_var(_env, "MSSQL_HOST_PORT", None, MSSQL_HOST_PORT)
-        _set_var(_env, "DRILL_HOST_PORT", None, DRILL_HOST_PORT)
```
Comment (shahar1) [inline dev/breeze/src/airflow_breeze/params/shell_params.py:546]: Fixing my ABC misorderings from the previous PRs

**Comment by potiuk** [comment]
Hmmm. I am not sure (to be honest) how this would work ... But I think I have little clarity myself what I wanted to do there..

Question: could you explain in your words what this task is about :D ? Maybe we can get to common understanding :D

**Comment by shahar1** [comment]
> Hmmm. I am not sure (to be honest) how this would work ... But I think I have little clarity myself what I wanted to do there..
> 
> Question: could you explain in your words what this task is about :D ? Maybe we can get to common understanding :D

I'll try to split it into two parts from my understanding:
1. In the first part of the original issue, the integration tests for Celery only utilize Redis as a broker and result backend. This PR addresses this issue by parameterizing the broker URL and running the same integration test for both Redis and Rabbitmq (it could be helpful for future alternatives if we decide to support them).

2. In the second part of the issue, you referred to a faulty caused by a version conflict between Celery and SQLAlchemy, and by creating an appropriate integration test, we could have discovered it in the CI. This PR doesn't address this issue (maybe it's a good time for a disclosure that I haven't touched the more "core" parts of Airflow components yet, so I might need more time to learn this area).

I'd be happy if you could merge this one, and create a separate issue for the second part (if you think that it's still relevant and worth implementing).

**Comment by potiuk** [comment]
Ah... Now I know what the problem with my understanding was ... Actually the tests already supported both redis and rabbitmq by the very same parameterisation - so your change just organizes it a bit better by converting it in a map and extracting the constants. 

There are two tests only that use the `_prepare_test_bodies` method and that did not change in your PR.

The idea however was to run **all** the integration tests for celery in similar way - once with rabbitmq and once with redis. Likely by running two different integrations "celery-rabbitmq" , "celery-redis" - basically running the same tests. This would require to change the marker logic a BIT I think - because that is not currently possible/easy.

**Comment by shahar1** [comment]
> Ah... Now I know what the problem with my understanding was ... Actually the tests already supported both redis and rabbitmq by the very same parameterisation - so your change just organizes it a bit better by converting it in a map and extracting the constants. 
> 
> There are two tests only that use the `_prepare_test_bodies` method and that did not change in your PR.
> 
> The idea however was to run **all** the integration tests for celery in similar way - once with rabbitmq and once with redis. Likely by running two different integrations "celery-rabbitmq" , "celery-redis" - basically running the same tests. This would require to change the marker logic a BIT I think - because that is not currently possible/easy. 
> 
> 
> 
> 

Gotcha, I'll try to dig deeper into that

**Comment by shahar1** [comment]
@potiuk So eventually, I thought that it would be easier and more precise to make integration tests for `redis` provider run separately (without `rabbitmq` running in the background), 
and make the tests for `celery` to import the `redis` service above using the [extension mechanism](https://docs.docker.com/compose/multiple-compose-files/extends/) so we won't have to duplicate code.
I updated the description in the PR so it would become clearer.

File: contributing-docs/testing/integration_tests.rst [hunk 2]
```diff
@@ -74,6 +74,8 @@ The following integrations are available:
 +--------------+----------------------------------------------------+
 | qdrant       | Integration required for Qdrant tests.             |
 +--------------+----------------------------------------------------+
+| redis        | *                                                  |
```
Comment (potiuk) [inline contributing-docs/testing/integration_tests.rst]: Should be a real description.

**Comment by potiuk** [comment]
Good idea. ❤️ it.  Just the descripton should be updated.

**Review by potiuk — APPROVED** [review]

**Comment by shahar1** [comment]
> Good idea. ❤️ it. Just the descripton should be updated.

Done :)

**Review by eladkal — APPROVED** [review]
Great work!

**Review by vincbeck — APPROVED** [review]

**Pull request merged**